libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -520,18 +520,21 @@ object MapboxRouteLineUtils {
             // which results in a distance of 0.0 in the distance array since.  The duplicated
             // coordinate seems to occur at the point the next leg begins. This distance item
             // is filtered out here.
+            val itemOmissionCount = leg.annotation()?.distance()?.count { it == 0.0 } ?: 0
             leg.annotation()?.distance()?.filter { it > 0.0 }?.forEachIndexed { index, distance ->
                 val percentDistanceTraveled = runningDistance / route.distance()
                 val isLegOrigin = index == 0
-                val isInRestrictedRange = restrictedRanges.any { it.contains(index) }
-                val isInAClosure = closureRanges.any { it.contains(index) }
+                val effectiveIndex = index + itemOmissionCount
```

Comment:
I guess it will be correct only if all zeros are located at the start of the distance list. 
I suggest doing something like this instead:
```kotlin
leg.annotation()?.distance()?.withIndex()?.filter { it.index > 0.0 }?.forEach { (index, distance) ->
    // use index as is
}
```
or
```kotlin
leg.annotation()?.distance()?.forEachIndexed { index, distance ->
    if (distance <= 0) return@forEachIndexed
    // use index as is
}
```
The second option doesn't create a filtered list, which may be more efficient.

Reply:
this is a good point, @cafesilencio can you add a test that covers this case?

Reply:
It's unlikely there would be more than entry with a zero distance but not necessarily impossible. I realized this last night so i'll make an adjustment to handle the anomaly.